### PR TITLE
IGNITE-19056 Added tests for RAFT node restart with the loss of log suffix, emulating disabled fsync situation

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-18475

### CODE REVIEW

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+                .changeNodeFinder().changeNetClusterNodes(
+                        range(port(0), port(NODES)).mapToObj(port -> "localhost:" + port).toArray(String[]::new)
+                )
+        );
+
+        assertThat(changeFuture, willCompleteSuccessfully());
+
+        nodes = range(0, NODES).mapToObj(SimpleIgniteNode::new).collect(toList());
+
+        nodes.forEach(SimpleIgniteNode::startService);
```
Comment (tkalkirill) [inline modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:130]: Shouldn't it be added to `cleanup`?
Can we wait for all nodes to start?

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+                    confManager.add(confEntry);
+                }
+            }
+
+            return super.init(opts);
+        }
+    }
+
+    private static class TestRaftGroupListener implements RaftGroupListener {
+        String lastValue;
```
Comment (tkalkirill) [inline modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:334]: Shouldn't it be `volatile` ?

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+        SimpleIgniteNode node0 = nodes.get(0);
+        SimpleIgniteNode node1 = nodes.get(1);
+        SimpleIgniteNode node2 = nodes.get(2);
+
+        assertThat(node1.getService().run(testWriteCommand("1")), willCompleteSuccessfully());
+        waitForValueReplication(nodes, "1");
+
+        node1.stopService();
+        node1.logStorage.truncateSuffix(1);
+        node1.startService();
```
Comment (tkalkirill) [inline modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:242]: Shouldn't you wait for the node to finish starting up?

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+
+        assertThat(node1.getService().run(testWriteCommand("1")), willCompleteSuccessfully());
+        waitForValueReplication(nodes, "1");
+
+        node1.stopService();
+        node2.stopService();
+
+        node1.logStorage.truncateSuffix(1);
+
+        node2.startService();
```
Comment (tkalkirill) [inline modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:270]: Shouldn't you wait for the node to finish starting up?

File: modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/storage/state/AbstractTxStateStorageTest.java [hunk 5]
```diff
@@ -117,7 +117,7 @@ public void testPutGetRemove() {
     }
 
     private List<ReplicationGroupId> generateEnlistedPartitions(int c) {
-        return IntStream.range(0, c).mapToObj(TestReplicationGroupId::new).collect(toList());
+        return IntStream.range(0, c).mapToObj(i -> new TestReplicationGroupId(Integer.toString(i))).collect(toList());
```
[the remark this answers is not in the document]
Comment (ibessonov) [reply modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/storage/state/AbstractTxStateStorageTest.java:120]: Maybe

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java (hunk 1 above)
Comment (ibessonov) [reply modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:130]: It's actually already added in the SimpleIgniteNode constructor

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java (hunk 2 above)
Comment (ibessonov) [reply modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:334]: I guess there is HB somewhere, but you're right, volatile should be explicit here

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java (hunk 3 above)
Comment (ibessonov) [reply modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:242]: No, following waiting of the log replication is enough

File: modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java (hunk 4 above)
Comment (ibessonov) [reply modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java:270]: Same reason, following `waitForValueReplication(nodes, "1");` is enough

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**